CMakeLists.txt

```diff
@@ -210,72 +100,31 @@ set (
     mungefs_ctl
     )
 
-file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/include)
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/include")
 
 add_custom_command(
-   OUTPUT ${CMAKE_BINARY_DIR}/include/${AVRO_FILE}.hpp
-   COMMAND LD_LIBRARY_PATH=${BOOST_PATH}/lib:${CLANG_LIBS} ${IRODS_EXTERNALS_FULLPATH_AVRO}/bin/avrogencpp -o ${CMAKE_BINARY_DIR}/include/${AVRO_FILE}.hpp -i ${CMAKE_SOURCE_DIR}/avro_schemas/${AVRO_FILE}.json
-   MAIN_DEPENDENCY ${CMAKE_SOURCE_DIR}/avro_schemas/${AVRO_FILE}.json
+   OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/include/${AVRO_FILE}.hpp"
+   COMMAND "${AVROCPP_AVROGEN_EXECUTABLE}" -o "${CMAKE_CURRENT_BINARY_DIR}/include/${AVRO_FILE}.hpp" -i "${CMAKE_CURRENT_SOURCE_DIR}/avro_schemas/${AVRO_FILE}.json"
+   MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/avro_schemas/${AVRO_FILE}.json"
```

Comment:
So the find module for avro and the new avro target is why we no longer have to use `LD_LIBRARY_PATH`, correct?

Reply:
Yes and no.

Externals will still need `LD_LIBRARY_PATH` for as long as our `avrogencpp` doesn't have a proper `RPATH`/`RUNPATH`, but it should be set before calling CMake, not *in* CMakeLists.txt.

Reply:
Makes sense.

I'm assuming externals will need a tweak to address the LD_LIBRARY_PATH-before-cmake then?

Reply:
Yes, among other things